test/IISIntegration.FunctionalTests/Utilities/IISApplication.cs

```diff
@@ -0,0 +1,173 @@
+            pool.ProcessModel.IdentityType = ProcessModelIdentityType.LocalSystem;
+            pool.ProcessModel.LoadUserProfile = true;
+        }
+
+        private void ConfigureSite(string contentRoot, int port)
+        {
+            var site = _serverManager.Sites.Add(WebSiteName, contentRoot, port);
+            site.Applications.Single().ApplicationPoolName = AppPoolName;
+        }
+
+        private void ConfigureAppHostConfig(string dllRoot)
+        {
+            var config = _serverManager.GetApplicationHostConfiguration();
+            var globalModulesSection = config.GetSection("system.webServer/globalModules");
+
+            var ancmFile = GetAncmLocation(dllRoot);
+            var found = false;
+            foreach (var element in globalModulesSection.GetCollection())
+            {
+                if ((string)element["name"] == @"AspNetCoreModuleV2")
+                {
+                    _originalAspNetCoreDllLocation = (string)element["image"];
+                    element["image"] = ancmFile;
+                    found = true;
+                    break;
+                }
+            }
+
+            // If ANCMV2 isn't in the global apphost.config, add it there.
+            if (!found)
+            {
+                _originalAspNetCoreDllLocation = "";
+                var addElement = globalModulesSection.GetCollection().CreateElement("add");
+                addElement["name"] = "AspNetCoreModuleV2";
+                addElement["image"] = ancmFile;
+                globalModulesSection.GetCollection().Add(addElement);
+            }
+        }
+
+        private string GetAncmLocation(string dllRoot)
+        {
+            var arch = _deploymentParameters.RuntimeArchitecture == RuntimeArchitecture.x64 ? @"x64\aspnetcorev2.dll" : @"x86\aspnetcorev2.dll";
+            var ancmFile = Path.Combine(dllRoot, arch);
+            if (!File.Exists(Environment.ExpandEnvironmentVariables(ancmFile)))
+            {
+                ancmFile = Path.Combine(dllRoot, "aspnetcorev2.dll");
+                if (!File.Exists(Environment.ExpandEnvironmentVariables(ancmFile)))
+                {
+                    throw new FileNotFoundException("AspNetCoreModule could not be found.", ancmFile);
+                }
+            }
+
+            return ancmFile;
+        }
+
+        public void StopAndDeleteAppPool()
+        {
+            if (string.IsNullOrEmpty(WebSiteName))
+            {
+                return;
+            }
+
+            RemoveSite();
+
+            RemoveAppPool();
+
+            RemoveAppHostConfig();
+        }
+
+        private void RemoveAppPool()
+        {
+            var appPool = _serverManager.ApplicationPools.FirstOrDefault(pool => pool.Name == AppPoolName);
+            if (appPool != null)
+            {
+                appPool.Stop();
+                _serverManager.ApplicationPools.Remove(appPool);
+                _serverManager.CommitChanges();
```

Comment:
When adding the site and app pool, the code calls `_serverManager.CommitChanges()` once at the end.  But when removing the site and app pool, the code calls `_serverManager.CommitChanges()` after each operation.  These should be consistent, and I would change the removal code to call once at the end if it's sufficient.  Minimize calls to `CommitChanges()`.